src/overlays/actors/ovl_En_Pst/z_en_pst.c

```diff
@@ -31,51 +138,312 @@ const ActorInit En_Pst_InitVars = {
+    if (this->stateFlags & 7) {
+        if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+            this->stateFlags &= ~0x30;
+            if (player->exchangeItemId == EXCH_ITEM_LETTER_TO_KAFEI) {
+                this->stateFlags |= 0x10;
+                this->exchangeItemId = player->exchangeItemId;
+            } else if (player->exchangeItemId != EXCH_ITEM_NONE) {
+                this->stateFlags |= 0x20;
+                this->exchangeItemId = player->exchangeItemId;
+            }
+            // If Letter to Kafei is deposited, value is set to 2
+            this->isLetterToKafeiDeposited = EnPst_HandleLetterDay1(this);
+            SubS_UpdateFlags(&this->stateFlags, 0, 7);
+            this->behaviour = 0;
+            this->msgEventCallback = NULL;
+            this->stateFlags |= 0x40;
+            this->msgEventScript = EnPst_GetMsgEventScript(this, play);
+            this->actionFunc = EnPst_UpdateSchedule;
+            ret = true;
+        }
+    }
+    return ret;
+}
+
+s32 EnPst_UpdateFlagsSubs(EnPst* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    SubS_UpdateFlags(&this->stateFlags, 3, 7);
+    return true;
+}
 
-#endif
+s32 EnPst_ProcessScheduleOutput(EnPst* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    s32 ret = false;
 
-extern ColliderCylinderInit D_80B2C4B8;
-extern CollisionCheckInfoInit2 D_80B2C4E4;
+    this->stateFlags = 0;
 
-extern UNK_TYPE D_06001A80;
+    switch (scheduleOutput->result) {
+        case POSTBOX_SCH_AVAILABLE:
+            ret = EnPst_UpdateFlagsSubs(this, play, scheduleOutput);
+            break;
+        case POSTBOX_SCH_CHECKED_BY_POSTMAN:
+            ret = true;
+            break;
+    }
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2B830.s")
+void EnPst_HandleSchedule(EnPst* this, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2B874.s")
+void EnPst_FollowSchedule(EnPst* this, PlayState* play) {
+    static u8* sScheduleScripts[] = {
+        D_80B2C200, D_80B2C20C, D_80B2C218, D_80B2C224, D_80B2C230,
+    };
+    s16 params = this->actor.params;
+    ScheduleOutput scheduleOutput;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2B8F4.s")
+    if (!Schedule_RunScript(play, sScheduleScripts[params], &scheduleOutput) ||
+        ((this->scheduleResult != scheduleOutput.result) &&
+         !EnPst_ProcessScheduleOutput(this, play, &scheduleOutput))) {
+        this->actor.shape.shadowDraw = NULL;
+        this->actor.flags &= ~ACTOR_FLAG_1;
+        scheduleOutput.result = POSTBOX_SCH_NONE;
+    } else {
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+        this->actor.flags |= ACTOR_FLAG_1;
+    }
+    this->scheduleResult = scheduleOutput.result;
+    EnPst_HandleSchedule(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2B974.s")
+void EnPst_UpdateSchedule(EnPst* this, PlayState* play) {
```

Comment:
Based on `EnBjt` this would probably be best named as `EnPst_Talk`.

Reply:
Generally I don't want things called `Update` if they're actually actionFuncs, it's clearer if we mainly save `Update` for things that are called in `Update` or set to `actor.update` (with exceptions where appropriate, of course).

Reply:
I don't think `talk` really fits. I went with UpdateSchedule because with `EnPst_HandleLetterDay1` and `EnPst_HandleLetterDay2(this)` update the delivery date of the letter (altering the schedule). Not only that, but it goes along with the other names in the file (FollowSchedule, HandleSchedule, ProcessSchedule...). I suppose an alternative I can think of right now could be `SetupSchedule`?

Reply:
I'll defer to Elliptic since he knows more about the Message scripting system and named the similar function in EnBjt, however I think Talk makes sense when considering the CheckTalk function above. This function gets set as the action function only if the Actor has some kind talk request to process.  The other stuff in this function seems to be based on how the conversation goes with `func_8010BF58` which kind of hides the actual talking going on.

Reply:
msgevent seems to be primarily a way to script conversations, with the ability to set other stuff (flags etc) at the same time.